homeassistant/components/asuswrt/config_flow.py

```diff
@@ -117,29 +122,41 @@ async def _async_check_connection(user_input: dict[str, Any]) -> str:
 
         except OSError:
             _LOGGER.error("Error connecting to the AsusWrt router at %s", host)
-            return RESULT_CONN_ERROR
+            return RESULT_CONN_ERROR, None
 
         except Exception:  # pylint: disable=broad-except
             _LOGGER.exception(
                 "Unknown error connecting with AsusWrt router at %s", host
             )
-            return RESULT_UNKNOWN
+            return RESULT_UNKNOWN, None
 
         if not api.is_connected:
             _LOGGER.error("Error connecting to the AsusWrt router at %s", host)
-            return RESULT_CONN_ERROR
+            return RESULT_CONN_ERROR, None
 
+        label_mac = await get_nvram_info(api, LABEL_MAC)
         conf_protocol = user_input[CONF_PROTOCOL]
         if conf_protocol == PROTOCOL_TELNET:
             api.connection.disconnect()
-        return RESULT_SUCCESS
+
+        unique_id = None
+        if label_mac and "label_mac" in label_mac:
+            unique_id = format_mac(label_mac["label_mac"])
+        return RESULT_SUCCESS, unique_id
 
     async def async_step_user(
         self, user_input: dict[str, Any] | None = None
     ) -> FlowResult:
         """Handle a flow initiated by the user."""
-        if self._async_current_entries():
-            return self.async_abort(reason="single_instance_allowed")
+
+        # if exist one entry without unique ID, we abort config flow
+        for unique_id in self._async_current_ids():
+            if unique_id is None:
+                _LOGGER.warning(
```

Comment:
Add a new abort reason and put this message in there. That way it's visible to users in the UI .

Reply:
Done.

Reply:
We can remove this warning being logged. It's already shown to the user.

Reply:
Warning removed.